src/context.cc

```diff
@@ -25,40 +25,21 @@
 #include "src/shared_data.h"
 #include "src/shared_queue.h"
 
-#define CHECK_FAIL(_call, _stream_type, _return_open, _return_closed)                              \
+#define CHECK_FAIL(_stream_type, _stream_type2, _return_open, _return_closed)                      \
   if (isFailed()) {                                                                                \
     if (plugin_->fail_open_) {                                                                     \
       return _return_open;                                                                         \
     } else {                                                                                       \
       failStream(_stream_type);                                                                    \
+      failStream(_stream_type2);                                                                   \
```

Comment:
Is there anything preventing this from being called multiple times?

Reply:
I guess there's no such a thing and it depends on the host implementation. So I'm going to add some guard.

Reply:
maybe we could leverage `destroyed_` used in `destroy()`.. WDYT?

Reply:
nvm, we shouldn't..